src/controller/TypedReadCallback.h

```diff
@@ -115,6 +115,14 @@ class TypedReadAttributeCallback final : public app::ReadClient::Callback
                 chip::Platform::Delete<app::AttributePathParams>(&aReadPrepareParams.mpAttributePathParamsList[i]);
             }
         }
+
+        if (aReadPrepareParams.mpDataVersionFilterList != nullptr)
+        {
+            for (size_t i = 0; i < aReadPrepareParams.mDataVersionFilterListSize; i++)
+            {
+                chip::Platform::Delete<app::DataVersionFilter>(&aReadPrepareParams.mpDataVersionFilterList[i]);
```

Comment:
This makes absolutely no sense, except when `aReadPrepareParams.mDataVersionFilterListSize` is 1.  For any value > 1 we would be deleting a pointer into the middle of the buffer.

Now our one consumer in fact only has a single data version filter, so this happens to work.  But I think we should be honest with ourselves and just:

1) VerifyOrDie(mDataVersionFilterListSize == 1) when the list is nonzero.
2) Call `chip::Platform::Delete<app::DataVersionFilter>(aReadPrepareParams.mpDataVersionFilterList)`.

I just realized that `mpAttributePathParamsList` has the same problem.  Needs to be fixed there too.

Reply:
done, thanks